### PR TITLE
Feature/route lane change idl

### PR DESCRIPTION
**Pull request by lewisid**
# PR Details
## Description

As identified in CAR-2785: Create High Level Design for Basic TravelDONE, UI needs a simple topic to display what is the next lane change direction is (left/right) and how far it is until an upcoming lane change is required. This story is to develop this logic, which should leverage existing plugins to identify when the lane change will actually occur and to make this information available to the UI. It is understood that the lane change location cannot be known sooner than it is planned by the relevant plugin. Should be published as soon as it is known and updated as actual point of lane change approaches or shifts.

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [x] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by msmcconnell** [comment]
Per @snallamothu we will hold off on merging this change for zephyr and will target it to develop as we do not have time to verify before Friday and its not use case critical functionality.

**Review by msmcconnell — CHANGES_REQUESTED** [review]
Blocking till zephyr release per my earlier comment

**Comment by msmcconnell** [comment]
@aswath1 @lewisid does this replace Ash's earlier PR https://github.com/usdot-fhwa-stol/carma-platform/pull/1300 ? If so lets close the other PR

File: carma/launch/guidance.launch [hunk 1]
```diff
@@ -108,8 +108,10 @@
   <include file="$(find pure_pursuit_jerk_wrapper)/launch/pure_pursuit_jerk_wrapper.launch" />
 
     <!-- Route Following Plugin -->
-  <include file="$(find route_following_plugin)/launch/route_following_plugin.launch" />
-  
+  <group>
+    <include file="$(find route_following_plugin)/launch/route_following_plugin.launch" />
+    <remap from="upcoming_lane_change_status" to="$(optenv CARMA_GUIDE_NS)/upcoming_lane_change_status"/>
```
Comment (msmcconnell) [inline carma/launch/guidance.launch:113]: I believe this remapping will not work. remappings are applied top-down within scope. The remappning is scoped to the group tag but there is nothing below it, so it has no impact on the system.

**Review by msmcconnell — CHANGES_REQUESTED** [review]

**Review by msmcconnell — CHANGES_REQUESTED** [review]

**Comment by MishkaMN** [comment]
Closed this PR in favor of https://github.com/usdot-fhwa-stol/carma-platform/pull/1320

**Pull request closed without merging**